addons/website_event/models/website_menu.py

```diff
@@ -28,3 +28,15 @@ def unlink(self):
                 event.write(dict((fname, False) for fname in to_update))
 
         return res
+
+    @api.model
+    def save(self, website_id, data):
+        for menu in data['data']:
+            if isinstance(menu['id'], str):
+                website_event_menu = self.env['website.event.menu'].search([('menu_id.parent_id', '=', menu['parent_id'])])
+                if website_event_menu:
+                    event_url = website_event_menu.event_id.website_url
+                    page_name = menu['url'].lstrip('/')
+                    menu['url'] = f"{event_url}/page/{page_name}"
```

Comment:
Hello @yagp-odoo 

I'm going to green the task to move on a bit but I'm still not sure about formatting automatically the url. This seems weird to me because the field name on the menu creation is `URL or Email` and not `Page name` and if someone create a new menu and write a URL like `/event/test-event-42/page/new_page` and then save, he will end with a page with the url `/event/test-event-42/page/event/test-event-42/page/new_page`.

I understand that we need to ensure that the page is linked to the correct event. So perhaps we just need to add an info/help message when we try to create a new website event menu/page or check that the event url is not already contains in it before adding it. Probably a discussion to have with the PO about that.

Cheers!

Reply:
Hi, @jeh-odoo 

>if someone create a new menu and write a URL like /event/test-event-42/page/new_page and then save, he will end with a page with the url /event/test-event-42/page/event/test-event-42/page/new_page.

I believe this `if website_event_menu and not menu['url'].startswith(event_url):` can solve this problem..?
Although submenu part you said in lognote remains, as we changed url value at save time, and the parent_id of submenu will differ from actual event menu so...

Thanks for validation :smile: